app/data/repositories/UsersRepository.py

```diff
@@ -1,68 +1,101 @@
-from app.data import databaseConnection
-from app.data.models import User
+from app.data import dbConn
+from app.data.models.User import User
 
-conn = databaseConnection.connect()
-# TODO try to return dic instead of array OR return User object?
-def fetchUsers():
-    query = "SELECT * FROM user"
+
+conn = dbConn.get()
+
+def create(name, email, password, birthday):
+
+        
+    name = name
+    email = email
+    password = password
+    birthday = birthday
+    
+    query = f"""INSERT INTO user (name, email, password, date_of_birth) 
+                    VALUES ('{name}', '{email}', '{password}', '{birthday}');"""
 
     c = conn.cursor()
 
     c.execute(query)
 
-    return c.fetchall()
+    conn.commit()
 
-def fetchUserById(user_id):
-    query = f"SELECT * FROM user WHERE user_name = '{user_id}';"
+    if c.lastrowid:
+        return True
+    
+    return False
+
+
+def user(data):
+    return User.user(data)
+
+
+def getAll():
+    query = "SELECT * FROM user"
 
     c = conn.cursor()
 
     c.execute(query)
     
-    return c.fetchone()
+    users = []
+
+    for user_data in c.fetchall():
+        
+        users.append(user(user_data))
+
+    return users
 
 
-def fetchUserByName(username):
+def getByEmailUsername():
+    return "huj"
+
     
-    query = f"SELECT * FROM user WHERE user_name = '{username}';"
+def getById(user_id):
+    query = f"SELECT * FROM user WHERE user_name = '{user_id}';"
 
     c = conn.cursor()
 
     c.execute(query)
     
-    user = User(c.fetchone())
+    return user(c.fetchone())
 
-    if user.ifUserExists():
-        return user
 
-    return None
+def getByName(username):
+    
+    query = f"SELECT * FROM user WHERE user_name = '{username}';"
+
+    c = conn.cursor()
 
+    c.execute(query)
+    
+    return user(c.fetchone())
 
-def fetchUserByEmail(email):
+# WHAT RETURN WHEN THERE IS NO USER IN DATABASE
+def getByEmail(email):
 
     query = f"SELECT * FROM user WHERE user_email = '{email}';"
 
     c = conn.cursor()
 
     c.execute(query)
 
-    return c.fetchone()
+    dbUser = c.fetchone()
 
+    if dbUser:
+        return user(dbUser)
 
-def saveUser(user: User):
-    
-    userName = user.username
-    userEmail = user.email
-    userPassword = user.password
-    birthDate = user.birthDay
+    return False
```

Comment:
return None, and check if you can set that method to return User or some optional of User

Reply:
I will do this in the next PR